### PR TITLE
Update rmse, ssim, top_k_categorical_accuracy in test for generating data with different rank

### PR DESCRIPTION
**Pull request by puhuk**
Description: Update `rmse`, `ssim`, `top_k_categorical_accuracy`

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/metrics/test_top_k_categorical_accuracy.py [hunk 1]
```diff
@@ -63,17 +63,17 @@ def _test_distrib_integration(device):
...
     def _test(n_epochs, metric_device):
         n_iters = 100
-        s = 16
+        batch_size = 16
         n_classes = 10
 
-        offset = n_iters * s
-        y_true = torch.randint(0, n_classes, size=(offset * idist.get_world_size(),)).to(device)
-        y_preds = torch.rand(offset * idist.get_world_size(), n_classes).to(device)
+        offset = n_iters * batch_size
```
Comment (sadra-barikbin) [inline tests/ignite/metrics/test_top_k_categorical_accuracy.py:69]: Here as well.

**Review by sadra-barikbin — CHANGES_REQUESTED** [review]

File: tests/ignite/metrics/test_top_k_categorical_accuracy.py (hunk 1 above)
Comment (sadra-barikbin) [reply tests/ignite/metrics/test_top_k_categorical_accuracy.py:69]: May be you've forgotten to add `rank` to `manual_seed` call in line 62

File: tests/ignite/metrics/test_ssim.py [hunk 2]
```diff
@@ -131,17 +131,17 @@ def _test_distrib_integration(device, tol=1e-4):
 
     rank = idist.get_rank()
     n_iters = 100
-    s = 10
-    offset = n_iters * s
+    batch_size = 10
+    offset = n_iters * batch_size
```
[the remark this answers is not in the document]
Comment (sadra-barikbin) [reply tests/ignite/metrics/test_ssim.py:135]: Maybe you've forgotten to call `torch.manual_seed(12 + rank)`.

File: tests/ignite/metrics/test_top_k_categorical_accuracy.py [hunk 3]
```diff
@@ -100,7 +98,8 @@ def update(engine, i):
     metric_devices = ["cpu"]
     if device.type != "xla":
         metric_devices.append(idist.device())
-    for _ in range(3):
+    for i in range(3):
```
Comment (sadra-barikbin) [inline tests/ignite/metrics/test_top_k_categorical_accuracy.py:101]: So you could bring line 61 near here.

Comment (sadra-barikbin) [reply tests/ignite/metrics/test_top_k_categorical_accuracy.py:99]: Could you please put it outside the `for` loop?

**Review by sadra-barikbin — APPROVED** [review]

File: tests/ignite/metrics/test_ssim.py [hunk 4]
```diff
@@ -130,18 +130,18 @@ def _test_distrib_integration(device, tol=1e-4):
...
-        y_pred = torch.rand(offset * idist.get_world_size(), 3, 28, 28, dtype=torch.float, device=device)
+        y_pred = torch.rand(n_iters * batch_size, 3, 28, 28, dtype=torch.float, device=device)
         y = y_pred * 0.65
 
         def update(engine, i):
             return (
-                y_pred[i * s + offset * rank : (i + 1) * s + offset * rank],
-                y[i * s + offset * rank : (i + 1) * s + offset * rank],
+                y_pred[i * batch_size : (i + 1) * batch_size],
+                y[i * batch_size : (i + 1) * batch_size],
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_ssim.py:144]:
```suggestion
                y_pred[i * batch_size : (i + 1) * batch_size, ...],
                y[i * batch_size : (i + 1) * batch_size, ...],
```

File: tests/ignite/metrics/test_top_k_categorical_accuracy.py [hunk 5]
```diff
@@ -58,22 +58,18 @@ def top_k_accuracy(y_true, y_pred, k=5, normalize=True):
...
-        y_preds = torch.rand(offset * idist.get_world_size(), n_classes).to(device)
+        y_true = torch.randint(0, n_classes, size=(n_iters * batch_size,)).to(device)
+        y_preds = torch.rand(n_iters * batch_size, n_classes).to(device)
 
         def update(engine, i):
             return (
-                y_preds[i * s + rank * offset : (i + 1) * s + rank * offset, :],
-                y_true[i * s + rank * offset : (i + 1) * s + rank * offset],
+                y_preds[i * batch_size : (i + 1) * batch_size],
+                y_true[i * batch_size : (i + 1) * batch_size, :],
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_top_k_categorical_accuracy.py:72]:
```suggestion
                y_true[i * batch_size : (i + 1) * batch_size],
```

**Review by vfdev-5 — APPROVED** [review]

**Pull request merged**